pkg/controller/rmdworkload/rmdworkload_controller.go

```diff
@@ -274,6 +248,50 @@ func (r *ReconcileRmdWorkload) findTargetedNodes(request reconcile.Request, rmdN
 	return targetedNodes, nil
 }
 
+func (r *ReconcileRmdWorkload) findRemovedNodes(request reconcile.Request, rmdNodeStates *intelv1alpha1.RmdNodeStateList, rmdWorkload *intelv1alpha1.RmdWorkload) (map[string]string, error) {
+	// Perform final check on RMD workloads vs rmdWorkload.Spec.Nodes to find any
+	// Nodes which may have been removed from the rmdWorkload since last reconcile.
+	// In this event, append node address and workload address to a map which will be returned.
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	removedNodes := make(map[string]string)
+	rmdWorkloadName := rmdWorkload.GetObjectMeta().GetName()
+
+	for _, rmdNodeState := range rmdNodeStates.Items {
+		// Get node service address
+		address, err := r.getPodAddress(rmdNodeState.Spec.Node, rmdNodeState.GetObjectMeta().GetNamespace())
+		if err != nil {
+			reqLogger.Error(err, "Failed to get pod address")
+			return nil, err
+		}
+
+		activeWorkloads, err := r.rmdClient.GetWorkloads(address)
+		if err != nil {
+			reqLogger.Info("Could not GET workloads.", "Error:", err)
+			return nil, err
+		}
+
+		workload := rmd.FindWorkloadByName(activeWorkloads, rmdWorkloadName)
+		if workload.UUID == rmdWorkloadName {
+			nodeFound := false
```

Comment:
I think we should change `nodeFound` to `nodeExistsOnRmdWorkloadSpec` to be clear what's happening